### PR TITLE
feat: add issuance of anoncreds credentials

### PR DESCRIPTION
**Pull request by TimoGlastra**
Adds issuance of anoncreds credentials. Have been working on this in the plane to IIW ✈️ 

See video for flow. Left is dev version of Siera Desktop, right one is the last release.

https://user-images.githubusercontent.com/23165168/234006134-9479ac95-c87c-4277-9eae-a0d69bd3b72f.mov

The flow isn't optimal yet, but most of the things will be fixed easily when we update to 0.4.x, mainly:
- Allow to use schemaId when creating a cred def not in the wallet (currently issue with seqNo, will be solved)
- No way to create new dids. We now create a did on bcovrin:test on agent creation. in the future we can add a did registrar, and allow the user to choose which did to use
- Only anoncreds flow supported (JSON-LD can be added in a follow up PR) 
- Netwrok is hardcoded to Indy (BCovrin Test) in the ui, as that's the only supported network. In 0.4.0 we can extract it based on the issuerId

### CODE REVIEW

**Review by Tommylans — APPROVED** [review]
Sick 🚀.

I think some design stuff needs to change but I let @janrtvld decide on that. Everything else LGTM and nice to see that you used the existing components 😄.

File: packages/siera-ui/src/components/credential-definitions/AnonCredsCredentialDefinitionTable.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                    {record.tag}
+                  </Text>
+                </td>
+                <td>
+                  <Text size="sm" weight={500}>
+                    {record.schemaId}
+                  </Text>
+                </td>
+                <td>
+                  {/* We can extract this once we update to 0.4.0. */}
```
Comment (janrtvld) [inline packages/siera-ui/src/components/credential-definitions/AnonCredsCredentialDefinitionTable.tsx:61]: Maybe make this a TODO/FIXME so we won't forget.

**Comment by janrtvld** [comment]
Some notes on the UI:

- Would it be an option to put this in the credentials tab and not in a separate AnonCreds tab? Or are there reasons to have a separate tab for just AnonCreds credentials?
- We can also make it so you press credentials in the side navigation, and it folds out to show three subtabs: 'Credentials', 'Schemas', and 'Definitions'. 
- As you reused a lot of the components i think the styling looks great! Make check if everything looks okay in the light mode as well.

File: packages/siera-ui/src/modals/IssueCredentialModal.tsx [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+  const credentialFormats =
+    form.values.protocolVersion === 'v1'
+      ? [
+          { value: 'anoncreds', label: 'AnonCreds' },
+          // JSON-LD not supported in v1 of protocol
+          { value: 'jsonld', label: 'JSON-LD', disabled: true },
+        ]
+      : [
+          { value: 'anoncreds', label: 'AnonCreds' },
+          // JSON-LD not yet implemented for v2 (but is supported)
```
Comment (janrtvld) [inline packages/siera-ui/src/modals/IssueCredentialModal.tsx:90]: Maybe add a tooltip to explain that it's coming in the future.

**Comment by TimoGlastra** [comment]
Action points:
- remove jsonld vs anoncreds as only AnonCreds is supported
- rename AnonCreds tab to templates

**Comment by blu3beri** [comment]
@TimoGlastra would it be possible to fix those action points and get this merged? The action points seem rather small so it would be good to just pick those up.

**Comment by blu3beri** [comment]
@TimoGlastra ^^

**Comment by TimoGlastra** [comment]
I don't really have time to finish this sadly. :( 

feel free to pick it up if you have the time

**Comment by blu3beri** [comment]
@Tommylans Is this something you might have time for? Otherwise I can also take a look. The changes seem quite trivial.

**Comment by Tommylans** [comment]
image

**Pull request merged**